### PR TITLE
Add perl readkey package

### PR DESCRIPTION
**Pull request by JL2210**
Added perl_read_key to Packages.

### CODE REVIEW

**Comment by uberhacker** [comment]
Please see xsltproc package comments.

**Comment by uberhacker** [comment]
Please remove the changes to dpkg and gettext packages on this PR.

File: packages/perl_read_key.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  source_sha256 '0fa4fb2f8145e3fb2c2129ad28d55be175abcc258f239ba8ddc2cd83790aa8fb'
+
+  def self.build
+    system "perl", "Makefile.PL"
+    system "make"
+    system "perl", "ReadKey_pm.PL"
+  end
+
+  def self.install
+    system "make", "install"
```
Comment (uberhacker) [inline packages/perl_read_key.rb]: Please add `DESTDIR=#{CREW_DEST_DIR}`.

**Comment by JL2210** [comment]
I need some help on this one.

**Comment by uberhacker** [comment]
You need help removing the dpkg and gettext package changes?

**Comment by JL2210** [comment]
No. there should be another file that perl should run inside the source directory, but it requires the perl module that it builds to be installed before it can run.

**Comment by JL2210** [comment]
But that too.

**Comment by JL2210** [comment]
Should I make another package?

**Comment by JL2210** [comment]
Is there in option for doing something between self.build and self.install?

**Comment by uberhacker** [comment]
Based on their install notes, I believe this is all you need?
```
To install, unpack somewhere, type "perl Makefile.PL", and then "make test".
If the compilation and the tests are successful, then change to root and run
"make install".
```
I would suggest you just create a new branch and recommit.
```
$ cd /path/to/packages
$ cp perl_read_key.rb ~/
$ git push origin --delete add-perl-readkey
$ git branch -D add-perl-readkey
$ git remote add chromebrew git@github.com:skycocker/chromebrew.git
$ git fetch chromebrew
$ git checkout chromebrew/master
$ git checkout -b add-perl-readkey
$ cp ~/perl_read_key.rb .
... make changes ...
$ git add perl_read_key.rb
$ git commit -m "Add perl_read_key package"
$ git push origin master
```

**Comment by uberhacker** [comment]
You could add this:
```
def self.check
  system 'make test'
end
```

**Comment by JL2210** [comment]
I think it would be better to create another package. Perl readkey requires PerlReadkey_pm to be run before it adds the module to perl. I believe this is an issue on the source code maker's part.

File: packages/perl_read_key_pm.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+require 'package'
+
+class Perl_read_key_pm < Package
+  description 'Character mode terminal access for Perl Term::ReadKey'
+  homepage 'https://metacpan.org/source/JSTOWE/TermReadKey-2.37/'
+  version '2.37'
+  source_url 'https://github.com/jonathanstowe/TermReadKey/archive/v2.37.tar.gz'
+  source_sha256 '0fa4fb2f8145e3fb2c2129ad28d55be175abcc258f239ba8ddc2cd83790aa8fb'
+
```
Comment (uberhacker) [inline packages/perl_read_key_pm.rb]: `depends_on 'perl_read_key'`?

File: packages/perl_read_key.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def self.install
+    system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+  end
+
+  def self.postinstall
+    puts "Please install perl_read_key_pm after installing this package,"
+    puts "This just fills in dependencies so it can compile correctly."
+    puts "If you are currently installing perl_read_key_pm, you are OK."
```
Comment (uberhacker) [inline packages/perl_read_key.rb]: Would adding `depends_on 'perl_read_key'` to perl_read_key_pm.rb remove the need for this message?  Also, you are missing an `end`. :)  Please test to make sure these actually install before creating the PR.  Please make all your commits locally until you are ready.

File: packages/perl_read_key_pm.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    system "perl", "Makefile.PL"
+    system "make", "uninstall"
+    system "perl", "ReadKey_pm.PL"
+    system "perl", "Makefile.PL"
+    system "make"
+    system "crew", "remove", "perl_read_key"
+  end
+
+  def self.check
+    system "make", "test"
```
Comment (uberhacker) [inline packages/perl_read_key_pm.rb]: Missing `end`.  Please, please test first. :)

File: packages/perl_read_key_pm.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+class Perl_read_key_pm < Package
+  description 'Character mode terminal access for Perl Term::ReadKey'
+  homepage 'https://metacpan.org/source/JSTOWE/TermReadKey-2.37/'
+  version '2.37'
+  source_url 'https://github.com/jonathanstowe/TermReadKey/archive/v2.37.tar.gz'
+  source_sha256 '0fa4fb2f8145e3fb2c2129ad28d55be175abcc258f239ba8ddc2cd83790aa8fb'
+
+  def self.build
+    system "crew", "install", "perl_read_key"
```
Comment (uberhacker) [inline packages/perl_read_key_pm.rb]: You don't need this if you use `depends_on 'perl_read_key'`.

**Comment by JL2210** [comment]
How do you make a puts line output blue text?

File: packages/perl_read_key_pm.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    system "perl", "Makefile.PL"
+    system "make", "uninstall"
+    system "perl", "ReadKey_pm.PL"
+    system "perl", "Makefile.PL"
+    system "make"
+    system "crew", "remove", "perl_read_key"
```
Comment (uberhacker) [inline packages/perl_read_key_pm.rb]: Why do you need to remove?

**Comment by JL2210** [comment]
depends_on perl read key will make perl_read_key install every time there is an update. The files are the same, it is just a messy fix that I could not find another workaround for. Also, if the files are the same, make install would

**Comment by JL2210** [comment]
fail because it cannot overwrite the files.

**Comment by uberhacker** [comment]
Maybe it would be better if you didn't separate the packages?  Just try and get it working in a single package, if possible.  To make the text blue, just add `.lightblue` at the end of the string.  There are other packages you can look at as an example.

**Comment by JL2210** [comment]
I'll try.

**Comment by JL2210** [comment]
Try it now

**Comment by uberhacker** [comment]
Sometimes the simplest way is the best. :)

**Pull request merged**